File: 0026_remove-duplicates-from-sorted-arra/solution.go
package _026_remove_duplicates_from_sorted_array


func removeDuplicates(nums []int) int {
	if len(nums) == 0 {
	    return 0
	}
	//计数器，也充当慢指针
	count := 0
	//j为快指针
	for j:=1; j <len(nums);j++{
		if nums[count] != nums[j]{
			count++
			nums[count] = nums[j]
		}

	}
	//返回计算后的长度
	return count + 1
}

File: 0026_remove-duplicates-from-sorted-arra/solution.py
from typing import List


class Solution:
    """
    时间复杂度：O(n)，假设数组的长度是 n，那么 i 和 j 分别最多遍历 n 步。
    空间复杂度：O(1)。
    """

    def removeDuplicates(self, nums: List[int]) -> int:
        if not nums:
            return 0
        i = 0
        for j in range(1, len(nums)):
            if nums[i] != nums[j]:
                i += 1
                nums[i] = nums[j]
        return i + 1
